libs/src/api/user.js

```diff
@@ -450,12 +578,100 @@ class Users extends Base {
+      UPLOAD_METADATA: 'UPLOAD_METADATA',
+      UPDATE_METADATA_ON_CHAIN: 'UPDATE_METADATA_ON_CHAIN',
+      ASSOCIATE_USER: 'ASSOCIATE_USER'
+    }
+    let phase = ''
+
+    const oldMetadata = this.userStateManager.getCurrentUser()
+    if (!oldMetadata) { throw new Error('No current user.') }
+
+    newMetadata = this._cleanUserMetadata(newMetadata)
+    this._validateUserMetadata(newMetadata)
+
+    // If the new metadata is the same as the original, it is a no-op and exit early
+    // NOTE: be careful -- this implies that the userStateManager is the source of truth
+    // if userStateManager is not up to date, can cause issues
+    const metadataFields = this._getMetadataFieldsToUpdate(newMetadata)
+    if (metadataFields.include.length === 0) { return }
+
+    try {
+      // Update user creator_node_endpoint on chain if applicable
+      if (newMetadata.creator_node_endpoint !== oldMetadata.creator_node_endpoint) {
+        phase = phases.UPDATE_CONTENT_NODE_ENDPOINT_ON_CHAIN
+        await this.contracts.UserFactoryClient.updateCreatorNodeEndpoint(userId, newMetadata['creator_node_endpoint'])
+        // Ensure DN has indexed creator_node_endpoint change
+        await this._waitForCreatorNodeEndpointIndexing(newMetadata.user_id, newMetadata.creator_node_endpoint)
+      }
+
+      // Upload new metadata object to CN
+      phase = phases.UPLOAD_METADATA
+      const { metadataMultihash, metadataFileUUID } = await this.creatorNode.uploadCreatorContent(newMetadata)
+
+      // Write metadata multihash to chain
+      phase = phases.UPDATE_METADATA_ON_CHAIN
+      const updatedMultihashDecoded = Utils.decodeMultihash(metadataMultihash)
+      const { txReceipt } = await this.contracts.UserFactoryClient.updateMultihash(userId, updatedMultihashDecoded.digest)
+
+      // Write remaining metadata fields to chain
+      const { latestBlockNumber } = await this._updateUserOperations(newMetadata, oldMetadata, userId, metadataFields.exclude)
+
+      // Write to CN to associate blockchain user id with updated metadata and block number
+      phase = phases.ASSOCIATE_USER
+      await this.creatorNode.associateCreator(userId, metadataFileUUID, Math.max(txReceipt.blockNumber, latestBlockNumber))
+
+      // Update libs instance with new user metadata object
+      this.userStateManager.setCurrentUser({ ...oldMetadata, ...newMetadata })
+    } catch (e) {
+      // TODO: think about handling the update metadata on chain and associating..
+      throw new Error(`_handleMetadata() Error -- Phase ${phase}: ${e}`)
+    }
+  }
+
+  /**
+  * Compares the original user state to an updated state. Determines the fields to include or exclude
+  * in the metadata update on chain.
+  * @param {Object} newMetadata fields to update in the current user state
+  */
+  _getMetadataFieldsToUpdate (newMetadata) {
+    const originalMetadata = this.userStateManager.getCurrentUser()
```

Comment:
also - i like the idea behind this, but it seems unrelated to the PR work. maybe i'm misunderstanding - can you explain why you added this logic?

Reply:
sure! so I added this fn to prevent any unnecessary uploads/updates to the metadata to the content nodes and on chain. this method compares the input `newMetadata` against what is in `userStateManager` for the props in `UPDATE_USER_PROPS`. If any prop value is different, add that prop to `include`. else, add to `exclude`. This way, when we make the chain update calls we can determine what to exclude.

Reply:
bumping earlier comment - you pull `this.userStateManager.getCurrentUser()` here when you should just pass it in from line 527

Reply:
and as for this new function as a whole - looking at `addUserOps()` and `udateUserOps`, they already diff metadata fields to only make chain calls for the fields that are changed, so this method seems redundant?

Reply:
just remove this method bc unnecessary